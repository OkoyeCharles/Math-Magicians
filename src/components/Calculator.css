.calculator__container {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 0.125em;
  width: max-content;

  background: #d3d3d3;
}

.calculator__container * {
  font-size: 1.25em;
}

.calculator__field {
  grid-column: 1/5;
  background: red;
}

.calculator__field > input {
  border: none;
  font-size: 2.5em;
  width: 100%;
  text-align: right;
  padding: .5em 0;
  outline: none;
  background: #858693;
}

.calculator__container > button {
  padding: 2em 3em;
  background: #e0e0e0;
  border: none;
}

.calculator__container > button:active {
  background: #ccc;
}

.calculator__container > button:nth-child(4n + 1), 
.calculator__container > button:last-child {
  background: #f4913e;
}

.calculator__container > button:nth-child(4n + 1):active,
.calculator__container > button:last-child:active {
  background: #d88038;
}

.calculator__container > .calc__zero {
  grid-column: 1/3;
}

/* Disable Calculator Input Arrows */
/* Chrome, Safari, Edge, Opera */
input::-webkit-outer-spin-button,
input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}
/* Firefox */
input[type=number] {
  -moz-appearance: textfield;
}