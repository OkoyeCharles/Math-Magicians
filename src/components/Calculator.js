import React, { useState } from 'react';
import calculate from '../logic/calculate';
import '../styles/Calculator.css';

function Calculator() {
  const [state, setState] = useState({ total: null, next: null, operation: null });

  const { total, next } = state;

  const handleClick = (e) => {
    const newState = calculate(state, e.target.innerText);
    setState(newState);
  };

  return (
    <div className="calculator__container">
      <div className="div calculator__field">
        <input type="number" defaultValue={next} disabled="disabled" />
        <input type="number" defaultValue={total} disabled="disabled" />
      </div>
      <button type="button" className="calc__AC" onClick={handleClick}>AC</button>
      <button type="button" className="calc__plusMinus" onClick={handleClick}>+/-</button>
      <button type="button" className="calc__percent" onClick={handleClick}>%</button>
      <button type="button" className="calc__divide" onClick={handleClick}>รท</button>
      <button type="button" className="calc__seven" onClick={handleClick}>7</button>
      <button type="button" className="calc__eight" onClick={handleClick}>8</button>
      <button type="button" className="calc__nine" onClick={handleClick}>9</button>
      <button type="button" className="calc__mult" onClick={handleClick}>x</button>
      <button type="button" className="calc__four" onClick={handleClick}>4</button>
      <button type="button" className="calc__five" onClick={handleClick}>5</button>
      <button type="button" className="calc__six" onClick={handleClick}>6</button>
      <button type="button" className="calc__minus" onClick={handleClick}>-</button>
      <button type="button" className="calc__one" onClick={handleClick}>1</button>
      <button type="button" className="calc__two" onClick={handleClick}>2</button>
      <button type="button" className="calc__three" onClick={handleClick}>3</button>
      <button type="button" className="calc__plus" onClick={handleClick}>+</button>
      <button type="button" className="calc__zero" onClick={handleClick}>0</button>
      <button type="button" className="calc__period" onClick={handleClick}>.</button>
      <button type="button" className="calc__equal" onClick={handleClick}>=</button>
    </div>
  );
}

export default Calculator;
