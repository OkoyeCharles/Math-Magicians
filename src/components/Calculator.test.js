import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from './Calculator';

describe('correctlly renders a component', () => {
  test('should render calculator component', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
