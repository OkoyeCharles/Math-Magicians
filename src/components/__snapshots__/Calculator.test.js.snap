// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctlly renders a component should render calculator component 1`] = `
<div
  className="calculator__container"
>
  <div
    className="div calculator__field"
  >
    <input
      defaultValue={null}
      disabled="disabled"
      type="number"
    />
    <input
      defaultValue={null}
      disabled="disabled"
      type="number"
    />
  </div>
  <button
    className="calc__AC"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="calc__plusMinus"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="calc__percent"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="calc__divide"
    onClick={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    className="calc__seven"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="calc__eight"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="calc__nine"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="calc__mult"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="calc__four"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="calc__five"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="calc__six"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="calc__minus"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="calc__one"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="calc__two"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="calc__three"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="calc__plus"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="calc__zero"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="calc__period"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="calc__equal"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;
